Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -1226,19 +1214,10 @@ std::unique_ptr<Update> TreeResolver::resolve()
         }
     }
 
-    if (m_hasUnresolvedAnchorPositionedElements) {
-        // During each style resolution, we need to iterate through all anchors
-        // to efficiently find the most recent anchor element in DOM tree order
-        // at any point in time during style resolution.
-        for (auto& anchorElement : m_anchorElements)
-            anchorElement->invalidateAncestorsForAnchor();
-
-        // We also need to ensure that style resolution visits any unresolved
-        // anchor-positioned elements.
-        for (auto elementAndState : m_anchorPositionedStateMap) {
-            if (!elementAndState.value->hasBeenResolved)
-                elementAndState.key.invalidateForResumingAnchorPositionedElementResolution();
-        }
+    for (auto element : m_anchorPositionedElementsToInvalidate) {
+        element->invalidateForResumingAnchorPositionedElementResolution();
+        if (element->renderer())
+            element->renderer()->setNeedsLayout();
```

Comment:
We should not mutate render tree from style resolver (by setting layout bits). Render tree updates go via RenderTreeUpdater.